packages/controllers/src/snaps/SnapController.ts

```diff
@@ -735,6 +809,130 @@ export class SnapController extends BaseController<
     }, this._idleTimeCheckInterval) as unknown as number;
   }
 
+  /**
+   * Checks all installed and unblocked snaps against the block list and blocks
+   * snaps as appropriate. See {@link SnapController.blockSnap} for more
+   * information.
+   */
+  async updateBlockedSnaps(): Promise<void> {
+    if (!this._checkSnapBlockList) {
+      throw new Error(BLOCK_LIST_UNDEFINED_ERROR);
+    }
+
+    const blockedSnaps = await this._checkSnapBlockList(
+      Object.values(this.state.snaps).reduce((blockListArg, snap) => {
+        blockListArg[snap.id] = snap.version;
+        return blockListArg;
+      }, {} as Record<SnapId, SemVerVersion>),
```

Comment:
You can use the generic `reduce<Record<SnapId, SemVerVersion>>` to eliminate casting.

Reply:
bd16c5d